src/DataProcessor.ts

```diff
@@ -92,47 +99,60 @@ export class DataProcessor implements IDataProcessor {
     // Makes the first wsman call
     const clientObj = this.clientManager.getClientObject(clientId)
     if ((clientObj.action === ClientAction.ADMINCTLMODE || clientObj.action === ClientAction.CLIENTCTLMODE) && !clientMsg.payload.digestRealm && !clientObj.activationStatus.missingMebxPassword) {
-      await this.amtwsman.batchEnum(clientId, '*AMT_GeneralSettings')
+      const messageId = (this.httpHandler.messageId++).toString()
+      const xmlRequestBody = this.amt.GeneralSettings(AMT.Methods.GET, messageId)
+      this.httpHandler.connectionParams = {
```

Comment:
hmm - im wondering if there is a better way to do this so as to not duplicate this in deactivate and activate. At the very least can set the connection parameters up in the construction and just change credentials?

Reply:
Resolved in deactivator PR